pandas/core/util/hashing.py

```diff
@@ -164,6 +164,29 @@ def hash_tuples(vals, encoding='utf8', hash_key=None):
     return h
 
 
+def hash_tuple(val, encoding='utf8', hash_key=None):
+    """
+    Hash a single tuple efficiently
+
+    Parameters
+    ----------
+    val : single tuple
+    encoding : string, default 'utf8'
+    hash_key : string key to encode, default to _default_hash_key
+
+    Returns
+    -------
+    hash
+
+    """
+    hashes = (hash_array(np.array([v]), encoding=encoding, hash_key=hash_key,
```

Comment:
Not sure whether `np.array([v])` will always do the correct coercion of the type. In the `hash_tuples` version, this coercion to dtyped arrays happens in `MultiIndex.from_tuples(vals)`

More in general, large part of the remaining time is spent in the `hash_array` function. A more specialized hasher for single scalars would probably further improve it